{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       ".container {\n",
       "    width: 100%;\n",
       "}\n",
       "div#notebook {\n",
       "    padding-top: 1em;\n",
       "}\n",
       "#header-container {\n",
       "    display: none;\n",
       "}\n",
       "#header-bar {\n",
       "    display: none;\n",
       "}\n",
       "#maintoolbar {\n",
       "    display: none;\n",
       "}\n",
       "#menubar-container {\n",
       "    position: fixed;\n",
       "    margin-top: 0;\n",
       "}\n",
       "#site {\n",
       "    height: auto !important;\n",
       "}\n",
       "</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run setup.ipynb\n",
    "%matplotlib inline\n",
    "import hapclust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((8296600, 773, 2), (8296600,))"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# obtain data from unphased callset - only needed for variant annotations\n",
    "callset = phase1_ar31.callset\n",
    "pos_all = allel.SortedIndex(callset['2L/variants/POS'])\n",
    "ann_all = callset['2L/variants/ANN'][:][['Annotation', 'HGVS_p']]\n",
    "\n",
    "# setup haplotype data\n",
    "callset_phased = phase1_ar31.callset_phased\n",
    "genotypes_phased = allel.GenotypeDaskArray(callset_phased['2L/calldata/genotype'])\n",
    "pos_phased = allel.SortedIndex(callset_phased['2L/variants/POS'])\n",
    "genotypes_phased.shape, pos_phased.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"allel allel-DisplayAs2D\"><span>&lt;HaplotypeArray shape=(1710, 1530) dtype=int8&gt;</span><table><thead><tr><th></th><th style=\"text-align: center\">0</th><th style=\"text-align: center\">1</th><th style=\"text-align: center\">2</th><th style=\"text-align: center\">3</th><th style=\"text-align: center\">4</th><th style=\"text-align: center\">...</th><th style=\"text-align: center\">1525</th><th style=\"text-align: center\">1526</th><th style=\"text-align: center\">1527</th><th style=\"text-align: center\">1528</th><th style=\"text-align: center\">1529</th></tr></thead><tbody><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">0</th><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">...</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td></tr><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">1</th><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">...</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td></tr><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">2</th><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">...</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td></tr><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">...</th><td style=\"text-align: center\" colspan=\"12\">...</td></tr><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">1707</th><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">...</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td></tr><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">1708</th><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">...</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td></tr><tr><th style=\"text-align: center; background-color: white; border-right: 1px solid black; \">1709</th><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">...</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td><td style=\"text-align: center\">0</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "<HaplotypeArray shape=(1710, 1530) dtype=int8>\n",
       "0 0 0 0 0 ... 0 0 0 0 0\n",
       "0 0 0 0 0 ... 0 0 0 0 0\n",
       "0 0 0 0 0 ... 0 0 0 0 0\n",
       "...\n",
       "0 0 0 0 0 ... 0 0 0 0 0\n",
       "0 0 0 0 0 ... 0 0 0 0 0\n",
       "0 0 0 0 0 ... 0 0 0 0 0"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# define general region we're going to analyse\n",
    "loc_region = pos_phased.locate_range(0, 4000000)\n",
    "pos_phased_region = pos_phased[loc_region]\n",
    "\n",
    "# chop genotypes to region, remove colony parents (8 samples) and turn into haplotype array\n",
    "gen_phased_region = genotypes_phased[loc_region][:, :-8].compute()\n",
    "haps = gen_phased_region.to_haplotypes()\n",
    "\n",
    "#chop into gene\n",
    "region_vgsc = SeqFeature('2L', 2358158, 2431617)\n",
    "loc = pos_phased_region.locate_range(region_vgsc.start, region_vgsc.end)\n",
    "h_vgsc = haps[loc]\n",
    "\n",
    "h_vgsc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def graph_haplotype_network(h,\n",
    "                            hap_colors='grey',\n",
    "                            distance_metric='hamming',\n",
    "                            network_method='mjn',\n",
    "                            comment=None,\n",
    "                            engine='neato',\n",
    "                            format='png',\n",
    "                            mode='major',\n",
    "                            overlap=True,\n",
    "                            splines=True,\n",
    "                            graph_attrs=None,\n",
    "                            node_size_factor=0.005,\n",
    "                            node_attrs=None,\n",
    "                            show_node_labels=False,\n",
    "                            fontname='monospace',\n",
    "                            fontsize=None,\n",
    "                            edge_length=0.5,\n",
    "                            edge_weight=10,\n",
    "                            edge_attrs=None,\n",
    "                            show_alternate_edges=True,\n",
    "                            alternate_edge_attrs=None,\n",
    "                            anon_width=0.03,\n",
    "                            anon_fillcolor='white',\n",
    "                            anon_node_attrs=None,\n",
    "                            intermediate_nodes=True,\n",
    "                            max_dist=5,\n",
    "                            variant_labels=None,\n",
    "                            debug=False,\n",
    "                            ):\n",
    "    \"\"\"TODO doc me\"\"\"\n",
    "\n",
    "    # check inputs\n",
    "    h = allel.HaplotypeArray(h)\n",
    "\n",
    "    # optimise - keep only segregating variants\n",
    "    ac = h.count_alleles()\n",
    "    loc_seg = ac.is_segregating()\n",
    "    h = h[loc_seg]\n",
    "    if variant_labels is not None:\n",
    "        variant_labels = np.asarray(variant_labels, dtype=object)[loc_seg]\n",
    "\n",
    "    # find distinct haplotypes\n",
    "    h_distinct_sets = h.distinct()\n",
    "\n",
    "    # find indices of distinct haplotypes - just need one per set\n",
    "    h_distinct_indices = [sorted(s)[0] for s in h_distinct_sets]\n",
    "\n",
    "    # reorder by index\n",
    "    ix = np.argsort(h_distinct_indices)\n",
    "    h_distinct_indices = [h_distinct_indices[i] for i in ix]\n",
    "    h_distinct_sets = [h_distinct_sets[i] for i in ix]\n",
    "\n",
    "    # obtain an array of distinct haplotypes\n",
    "    h_distinct = h.take(h_distinct_indices, axis=1)\n",
    "\n",
    "    # deal with colors - count how many of each color per distinct haplotype\n",
    "    color_counters = None\n",
    "    if isinstance(hap_colors, (list, tuple, np.ndarray)):\n",
    "        assert len(hap_colors) == h.n_haplotypes\n",
    "        color_counters = [\n",
    "            collections.Counter([hap_colors[i] for i in s])\n",
    "            for s in h_distinct_sets\n",
    "        ]\n",
    "\n",
    "    # count how many observations per distinct haplotype\n",
    "    hap_counts = [len(s) for s in h_distinct_sets]\n",
    "\n",
    "    # compute pairwise distance matrix\n",
    "    assert distance_metric in ['hamming', 'jaccard']\n",
    "    dist = allel.pairwise_distance(h_distinct, metric=distance_metric)\n",
    "    dist *= h.n_variants\n",
    "    dist = scipy.spatial.distance.squareform(dist).astype(int)\n",
    "\n",
    "    if network_method.lower() == 'mst':\n",
    "\n",
    "        # compute minimum spanning tree\n",
    "        edges = scipy.sparse.csgraph.minimum_spanning_tree(dist).toarray().astype(int)\n",
    "\n",
    "        # deal with maximum distance\n",
    "        if max_dist:\n",
    "            edges[edges > max_dist] = 0\n",
    "\n",
    "        # no alternate edges when using mst\n",
    "        alternate_edges = None\n",
    "\n",
    "    elif network_method.lower() == 'msn':\n",
    "\n",
    "        # compute network\n",
    "        edges, alternate_edges = minimum_spanning_network(dist, max_dist=max_dist, debug=debug)\n",
    "        edges = np.triu(edges)\n",
    "        alternate_edges = np.triu(alternate_edges)\n",
    "\n",
    "    elif network_method.lower() == 'mjn':\n",
    "\n",
    "        # compute network - N.B., MJN may add new haplotypes\n",
    "        h_distinct, edges, alternate_edges = median_joining_network(h_distinct, max_dist=max_dist, debug=debug)\n",
    "        edges = np.triu(edges)\n",
    "        alternate_edges = np.triu(alternate_edges)\n",
    "\n",
    "    else:\n",
    "        raise ValueError(network_method)\n",
    "\n",
    "    # setup graph\n",
    "    graph = graphviz.Digraph(comment=comment, engine=engine, format=format)\n",
    "    if graph_attrs is None:\n",
    "        graph_attrs = dict()\n",
    "    graph_attrs.setdefault('overlap', str(overlap).lower())\n",
    "    graph_attrs.setdefault('splines', str(splines).lower())\n",
    "    graph_attrs.setdefault('mode', mode)\n",
    "    graph_attrs.setdefault('sep', '0')\n",
    "    graph.attr('graph', **graph_attrs)\n",
    "\n",
    "    # add the main nodes\n",
    "    if node_attrs is None:\n",
    "        node_attrs = dict()\n",
    "    node_attrs.setdefault('fixedsize', 'true')\n",
    "    node_attrs.setdefault('shape', 'circle')\n",
    "    node_attrs.setdefault('fontname', fontname)\n",
    "    node_attrs.setdefault('fontsize', str(fontsize))\n",
    "    if anon_node_attrs is None:\n",
    "        anon_node_attrs = dict()\n",
    "    anon_node_attrs.setdefault('fixedsize', 'true')\n",
    "    anon_node_attrs.setdefault('shape', 'circle')\n",
    "    anon_node_attrs.setdefault('style', 'filled')\n",
    "    anon_node_attrs.setdefault('fillcolor', anon_fillcolor)\n",
    "    anon_node_attrs.setdefault('fontname', fontname)\n",
    "    anon_node_attrs.setdefault('fontsize', str(fontsize))\n",
    "    for i in range(edges.shape[0]):\n",
    "        kwargs = dict()\n",
    "\n",
    "        if i < len(hap_counts):\n",
    "            # original haplotype\n",
    "\n",
    "            n = hap_counts[i]\n",
    "\n",
    "            # calculate width from number of items - make width proportional to area\n",
    "            width = np.sqrt(n * node_size_factor)\n",
    "\n",
    "            # determine style and fill color\n",
    "            if color_counters:\n",
    "                cc = color_counters[i]\n",
    "                if len(cc) > 1:\n",
    "                    # more than one color, make a pie chart\n",
    "                    style = 'wedged'\n",
    "                    fillcolor = ':'.join(['%s;%s' % (k, v/n) for k, v in cc.items()])\n",
    "                else:\n",
    "                    # just one color, fill with solid color\n",
    "                    style = 'filled'\n",
    "                    fillcolor = list(cc.keys())[0]\n",
    "            else:\n",
    "                style = 'filled'\n",
    "                fillcolor = hap_colors\n",
    "\n",
    "            kwargs.update(node_attrs)\n",
    "            kwargs.setdefault('style', style)\n",
    "            kwargs.setdefault('fillcolor', fillcolor)\n",
    "            kwargs.setdefault('width', str(width))\n",
    "\n",
    "        else:\n",
    "            # not an original haplotype, inferred during network building\n",
    "\n",
    "            n = 1\n",
    "\n",
    "            width = anon_width\n",
    "            fillcolor = anon_fillcolor\n",
    "            kwargs.update(anon_node_attrs)\n",
    "            kwargs.setdefault('width', str(anon_width))\n",
    "\n",
    "        # add graph node\n",
    "        if show_node_labels is False:\n",
    "            label = ''\n",
    "        elif show_node_labels is True:\n",
    "            label = str(i)\n",
    "        elif isinstance(show_node_labels, int) and n >= show_node_labels:\n",
    "            label = str(i)\n",
    "        elif show_node_labels == 'count' and n > 1:\n",
    "            label = str(n)\n",
    "        else:\n",
    "            label = ''\n",
    "        kwargs.setdefault('label', label)\n",
    "        graph.node(str(i), **kwargs)\n",
    "\n",
    "    # setup defaults\n",
    "    if edge_attrs is None:\n",
    "        edge_attrs = dict()\n",
    "    edge_attrs.setdefault('style', 'normal')\n",
    "    edge_attrs.setdefault('weight', str(edge_weight))\n",
    "    edge_attrs.setdefault('fontname', fontname)\n",
    "    edge_attrs.setdefault('fontsize', str(fontsize))\n",
    "    edge_attrs.setdefault('dir', 'none')\n",
    "    if alternate_edge_attrs is None:\n",
    "        alternate_edge_attrs = dict()\n",
    "    alternate_edge_attrs.setdefault('style', 'dashed')\n",
    "    alternate_edge_attrs.setdefault('weight', str(edge_weight))\n",
    "    alternate_edge_attrs.setdefault('fontname', fontname)\n",
    "    alternate_edge_attrs.setdefault('fontsize', str(fontsize))\n",
    "    alternate_edge_attrs.setdefault('dir', 'none')\n",
    "\n",
    "    # add main edges\n",
    "    _graph_edges(graph,\n",
    "                 edges,\n",
    "                 hap_counts,\n",
    "                 node_size_factor,\n",
    "                 edge_length,\n",
    "                 anon_width,\n",
    "                 intermediate_nodes,\n",
    "                 edge_attrs,\n",
    "                 anon_node_attrs,\n",
    "                 h_distinct,\n",
    "                 variant_labels)\n",
    "\n",
    "    # add alternate edges\n",
    "    if show_alternate_edges and alternate_edges is not None:\n",
    "        _graph_edges(graph,\n",
    "                     alternate_edges,\n",
    "                     hap_counts,\n",
    "                     node_size_factor,\n",
    "                     edge_length,\n",
    "                     anon_width,\n",
    "                     intermediate_nodes,\n",
    "                     alternate_edge_attrs,\n",
    "                     anon_node_attrs,\n",
    "                     h_distinct,\n",
    "                     variant_labels)\n",
    "\n",
    "    return graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
